feat(curriculum): add pricing plan lab to FSD cert

mtrivera opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitHub Codespaces.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

<!-- Feel free to add any additional description of changes below this line -->

closes https://github.com/freeCodeCamp/CurriculumExpansion/issues/776

mtrivera committed:
build: Scaffold pricing-plans-lab project

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,30 @@
+---
+id: 68e45a2b4ab9b2f48b1cc6df
+title: Step 1
```
```suggestion
title: Design a Pricing Plans Layout Page
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,30 @@
+---
+id: 68e45a2b4ab9b2f48b1cc6df
+title: Step 1
+challengeType: 0
+dashedName: step-1
+# demoType can either be 'onClick' or 'onLoad'. If the project or lab doesn't have a preview, delete the property
+demoType: onClick
```
```suggestion
challengeType: 25
dashedName: design-a-pricing-plans-layout-page
demoType: onClick
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,30 @@
+
+```
+
+# --seed--
+
+## --seed-contents--
+
+```html
+--fcc-editable-region--
+
+--fcc-editable-region--
+```
````
````suggestion
# --seed--

## --seed-contents--

```html

```

```css

```

# --solutions--

```html

```

```css

```

````

mtrivera committed:
Apply suggestions from code review

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

mtrivera committed:
fix: Move order of blocks in json file

Put lab-pricing-plans-layout below lab-page-of-playing cards

mtrivera committed:
feat: Add user stories for lab-pricing-plans-layout

mtrivera committed:
Merge branch 'pricing-plan-lab-776' of github.com:mtrivera/freeCodeCamp into pricing-plan-lab-776

mtrivera committed:
Merge branch 'freeCodeCamp:main' into pricing-plan-lab-776

mtrivera committed:
Merge branch 'pricing-plan-lab-776' of github.com:mtrivera/freeCodeCamp into pricing-plan-lab-776

mtrivera committed:
feat: Add HTML seed code

mtrivera committed:
feat: Add HTML and CSS solutions

mtrivera committed:
feat: Add note to user stories

Advise user to link stylesheet in HTML and apply CSS

mtrivera committed:
test: Check h1 element for correct text

h1 element should exist and contain Pricing Plans text

mtrivera committed:
test: Check for three div elements and class

There should be three div elements with the pricing-card class

mtrivera committed:
test: Div has required classes

A div element has classes pricing-card and basic-plan

mtrivera committed:
test: Div has required classes

A div element has classes pricing-card and premium-plan

mtrivera committed:
test: Div has required classes

A div element has classes pricing-card and pro-plan

jdwilkin4 commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,272 @@
+
+```
+
+# --solutions--
````
When I view the demo project, the layout seems off here. 

<img width="924" height="623" alt="Screenshot 2025-10-14 at 4 06 01 PM" src="https://github.com/user-attachments/assets/4325c654-1d8d-4aab-a676-c6c5f07daa27" />

We should change it to something like this down below for larger screens. and then smaller screens can still be in a column

<img width="922" height="559" alt="Screenshot 2025-10-14 at 4 08 04 PM" src="https://github.com/user-attachments/assets/29c0b956-793b-4a4f-9077-d9a5decf0799" />

mtrivera replied:
Working on this now

jdwilkin4 commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,272 @@
+assert.equal(h1.textContent, 'Pricing Plans', 'The h1 has the text Pricing Plans');
+```
+
+You should have three `div` elements with the class `pricing-card`
````
```suggestion
You should have three `div` elements with the class `pricing-card`.
```

jdwilkin4 commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,272 @@
+assert.exists(div);
+```
+
+You should have a `div` element with the class `pricing-card` and `pro-plan`
````
```suggestion
You should have a `div` element with the class `pricing-card` and `pro-plan`.
```

jdwilkin4 reviewed (commented):
Start testing things locally and left some comments

mtrivera committed:
docs: Add intro and move above review-css-flexbox

mtrivera commented on client/i18n/locales/english/intro.json:
```diff
@@ -5019,6 +4986,10 @@
         "title": "Certified Full Stack Developer Exam",
         "intro": ["Pass this exam to become a Certified Full Stack Developer."]
+      },
+      "lab-pricing-plans-layout": {
```
See commit `fd2fabe`

mtrivera commented on client/i18n/locales/english/intro.json:
```diff
@@ -5019,6 +4986,10 @@
+      },
+      "lab-pricing-plans-layout": {
+        "title": "Design a Pricing Plans Layout Page",
+        "intro": ["Design a Pricing Plans Layout Page", ""]
```
See commit `fd2fabe`

mtrivera committed:
test: Check each pricing-card for flex styles

Flex styles: flex: 0 0 200px; flex-direction: column; display: flex, justify-content: space-between

mtrivera committed:
test: Check basic-plan is first element

mtrivera committed:
test: Check pro-plan for flex-grow and order

mtrivera committed:
test: Check premium-layout is last element

mtrivera committed:
style: Add periods to test statements

mtrivera committed:
build: Make available once PR merged

mtrivera commented on curriculum/structure/blocks/lab-pricing-plans-layout.json:
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "Design a Pricing Plans Layout Page",
+  "isUpcomingChange": true,
```
See commit `708384d`

mtrivera commented:
While I work on the layout issue, can you try my tests? I finished adding them (will clean up later, just need to verify the tests).

mtrivera committed:
fix: View card row layout on large screens

mtrivera committed:
test: Clean up tests and more robust

Remove exists checks. For pricing-card tests, test if actual div element before moving on

mtrivera committed:
style: Change basic-card to basic-plan

mtrivera committed:
Merge branch 'staging' of github.com:mtrivera/freeCodeCamp into pricing-plan-lab-776

jdwilkin4 committed:
fix: incorrect placement for intro.json file

jdwilkin4 committed:
fix: incorrect placement for intro.json file

jdwilkin4 reviewed (commented):
I pushed up a commit to move the intro to the correct spot in the `intro.json` file. Otherwise, the project would not run because it would have shown error messages

jdwilkin4 commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,312 @@
+const div = document.querySelector('.pricing-card.basic-plan');
+assert.strictEqual(div.tagName, 'DIV');
+assert.isTrue(div.classList.contains('basic-plan'), 'The div contains the basic-plan class');
+assert.isTrue(div.classList.contains('pricing-card'), 'The div contains the pricing-card class');
```
```suggestion
const div = document.querySelector('.pricing-card.basic-plan');

assert.strictEqual(div?.tagName, 'DIV');
assert.isTrue(div?.classList.contains('basic-plan'));
assert.isTrue(div?.classList.contains('pricing-card'));
```

jdwilkin4 commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,312 @@
+const helper = new __helpers.CSSHelp(document);
+assert.strictEqual(helper.getStyle('.pro-plan').getPropVal('order'), '1');
+assert.strictEqual(helper.getStyle('.pro-plan').getPropVal('flex-grow'), '2');
+```
````
this should be broken up into two sepeate tests and hints

- one test and hint for the order
- one test and hint for the flex grow

also the hint right now doesn't mention it should be set to 2. the directions say that but the hint doesn't

jdwilkin4 reviewed (commented):
Tested it locally and left some comments.

But also there will need to be some additional user stories and tests/hints added.

Right now it is possible to pass with only an h1 displaying on the page. 
It would be beneficial from an educational standpoint to have some other content so they can see the flex properties at work.

So the minimum they should be able to pass with should be this

<img width="637" height="220" alt="Screenshot 2025-10-16 at 2 32 33 PM" src="https://github.com/user-attachments/assets/2d2fc2c5-4445-479a-86f5-d48d4a282d9f" />

Here is my current code for that

```html
<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8" />
  <title>Pricing Plans Layout Page</title>
  <link rel="stylesheet" href="styles.css">
</head>

<body>
  <h1>Pricing Plans</h1>
  <div class="pricing-container">
    <div class="pricing-card basic-plan">
      <h2>Basic</h2>
      <p class="price">$19/month</p>
    </div>
    <div class="pricing-card premium-plan">
      <h2>Premium</h2>
      <p class="price">$29/month</p>
    </div>
    <div class="pricing-card pro-plan">
      <h2>Pro</h2>
      <p class="price">$29/month</p>
    </div>
  </div>
</body>

</html>
```

```css
.pricing-container{
  display:flex;
  flex-wrap:wrap;
}
.pricing-card{
  display:flex;
  flex-direction:column;
  justify-content:space-between;
  flex:0 0 200px;
}

.basic-plan{
  order:0;
}

.pro-plan{
  order:1;
  flex-grow:2;
}

.premium-plan{
  order:2;
}

```

Add user stories and hints/tests for the following:

- add a `pricing-container` element for the pricing cards
- set the pricing-container selector to flex and flex-wrap:wrap;
- check to make sure they have h2 elements with the appropriate text for each pricing card
- check to make sure they have p elements with prices for each pricing card

jdwilkin4 reviewed (commented):
It would be nice to also add a user story and hints/test for a border around the pricing-card

```css
.pricing-card {
  display:flex;
  flex-direction:column;
  justify-content:space-between;
  flex:0 0 200px;
  border:2px solid black;
}
```

That way campers can see how the pro plan takes up more space than the other items in the container 
<img width="843" height="231" alt="Screenshot 2025-10-16 at 2 42 29 PM" src="https://github.com/user-attachments/assets/386505f9-3acc-486f-bddc-b3d8a3280fcc" />

mtrivera committed:
test: Make tests more effective

Remove chai assert message arguements and add optional chaining operators

mtrivera committed:
Merge branch 'pricing-plan-lab-776' of github.com:mtrivera/freeCodeCamp into pricing-plan-lab-776

mtrivera committed:
test: Split up pricing-card test

mtrivera committed:
test: Split up the pro-plan test

mtrivera committed:
style: Add periods to pricing-card tests

mtrivera committed:
docs: Add user stories to see flex properties at work

mtrivera committed:
docs: Add pricing-card border user story

mtrivera committed:
docs: Refine pricing-container user story

mtrivera committed:
test: Add pricing-container tests

mtrivera committed:
test: Add basic-plan h2 and p tests

mtrivera committed:
test: Add pro-plan h2 and p tests

mtrivera committed:
test: Add premium-plan h2 and p tests

mtrivera committed:
fix: Change test values of basic and pro plan

mtrivera committed:
docs: Correct user story values for plan prices

mtrivera committed:
fix: Target h2 elements on assert tests

mtrivera committed:
feat: Add border to pricing-card for visibility

mtrivera committed:
test: Add pricing-card border test

mtrivera committed:
docs: Clarify premium-plan hint to include order and 2

mtrivera commented:
Made all the changes. The `Ready for Review` button is not visible.

jdwilkin4 commented:
@mtrivera 

> Made all the changes. The Ready for Review button is not visible.

That would only be visible if your PR was still in draft status. This PR was already out of draft status so the button no longer appears 👍🏾

mtrivera committed:
test: Add optional chain operator to h1 test

mtrivera committed:
test: Improve user story 2 part 1 test

Check that the div element is truthy first before running the test

mtrivera committed:
docs: Improve user story 1 hint grammar

mtrivera committed:
docs: Improve user story 2 hint grammar

mtrivera committed:
docs: Improve user story 2 part 1 hint grammar

mtrivera committed:
docs: Improve user story 2 part 2 hint grammar

mtrivera committed:
docs: Improve user story 3 part 1 hint grammar

mtrivera committed:
docs: Improve user story 3 part 2 hint grammar

mtrivera committed:
docs: Improve user story 3 part 3 hint grammar

mtrivera committed:
docs: Improve user story 10 part display flex grammar

mtrivera committed:
test: Refine user story 10 display flex test

mtrivera committed:
test: Refine user story 10 flex-direction test

mtrivera committed:
docs: Improve user story 10 flex-direction grammar

mtrivera committed:
docs: Improve user story 10 justify-content grammar

mtrivera committed:
test: Improve user story 10 justify-content test

mtrivera committed:
docs: Improve user story 10 flex grammar

mtrivera committed:
test: Check for truthy property flex user story 10

mtrivera committed:
docs: Improve user story 10 border grammar

mtrivera committed:
test: Check for truthy property border user story 10

mtrivera committed:
docs: Improve user story 11 grammar

mtrivera committed:
test: Improve user story 11 test

mtrivera committed:
docs: Improve user story 4 grammar

mtrivera committed:
test: Check h2 is truthy for user story 4 test

mtrivera committed:
docs: Improve user story 5 grammar

mtrivera committed:
test: Check p is truthy for user story 5 test

mtrivera committed:
docs: Improve user story 6 grammar

mtrivera committed:
test: Check h2 is truthy for user story 6 test

mtrivera committed:
docs: Improve user story 7 grammar

mtrivera committed:
test: Check p is truthy for user story 7

mtrivera committed:
test: Check flex-grow property is truthy for user story 12

mtrivera committed:
docs: Improve user story 12 hint grammar

mtrivera committed:
docs: Improve user story 12 flex-grow hint grammar

mtrivera committed:
docs: Improve user story 8 hint grammar

mtrivera committed:
test: Check h2 is truthy for user story 8

mtrivera committed:
docs: Improve user story 9 hint grammar

mtrivera committed:
test: Check p is truthy for user story 9

mtrivera committed:
docs: Improve user story 13 hint grammar

mtrivera committed:
test: Check div is truthy for user story 13

mtrivera committed:
docs: Change voice of user story 13 hint

mtrivera commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,419 @@
+assert.isTrue(div?.classList.contains('pricing-card'), 'The div contains the pricing-card class');
+```
+
+Your `.pricing-card` element should contain the `display` property and the `flex` value.
````
I came up with this: 
```js
const div = document.querySelector('.pricing-card');
assert.isOk(div);
const helper = new __helpers.CSSHelp(document);
assert.strictEqual(helper.getStyle('.pricing-card')?.display, 'flex');
```
Is this what you meant? Please advise.

ilenia-magoni replied:
I meant add a test "You should hacve a `.pricing-card` selector" that check that `new __helpers.CSSHelp(document).getStyle('.pricing-card')` finds the selector

mtrivera committed:
test: Check for truthy prop flex user story 10

mtrivera committed:
test: Check for truthy property border user story 10

mtrivera committed:
test: Check flex-grow property is truthy for user story 12

mtrivera committed:
test: Remove assert message from user story 3 part 2

mtrivera committed:
test: Check that pricing-card selector exists

mtrivera committed:
refactor: Clean up tests and make more robust

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+assert.strictEqual(h1?.textContent, 'Pricing Plans');
+```
+
+Your `div` element should contain the class `pricing-container`.
````
```suggestion
Your `div` element should have the class `pricing-container`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+assert.strictEqual(div?.tagName, 'DIV');
+```
+
+Your `pricing-container` element should contain the property `display` and the value `flex`.
````
```suggestion
Your `.pricing-container` selector should have the property `display` and the value `flex`.
```

you are testing specifically for the selector. To test that the element has a certain styling you would need to use `getComputedStyle`

https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+assert.lengthOf(divEls, 3);
+```
+
+Your `div` element should contain the class `pricing-card` and the class `basic-plan`.
````
```suggestion
Your first `div` element should have the class `pricing-card` and the class `basic-plan`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+assert.isTrue(div?.classList.contains('pricing-card'));
+```
+
+Your `div` element should contain the class `pricing-card` and the class `premium-plan`.
````
```suggestion
Your second `div` element should have the class `pricing-card` and the class `premium-plan`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,414 @@
+const div = document.querySelector('.pricing-card.premium-plan');
+
+assert.strictEqual(div?.tagName, 'DIV');
+assert.isTrue(div?.classList.contains('premium-plan'));
```
this is redundant, you are selecting the element with the class, so this and the line below will be true by default

you may want to get the element using a `div:nth-of-type()` selector for all three tests like this

mtrivera committed:
docs: Change the word 'contain' to 'have' in hints

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,414 @@
+
+assert.strictEqual(div?.tagName, 'DIV');
+assert.isTrue(div?.classList.contains('premium-plan'));
+assert.isTrue(div?.classList.contains('pricing-card'));
```
```suggestion
const div = document.querySelector('div:nth-of-type(2)');

assert.isTrue(div?.classList.contains('premium-plan'));
assert.isTrue(div?.classList.contains('pricing-card'));
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,414 @@
+
+assert.strictEqual(div?.tagName, 'DIV');
+assert.isTrue(div?.classList.contains('pro-plan'),);
+assert.isTrue(div?.classList.contains('pricing-card'));
```
```suggestion
const div = document.querySelector('div:nth-of-type(3)');

assert.isTrue(div?.classList.contains('pro-plan'),);
assert.isTrue(div?.classList.contains('pricing-card'));
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+assert.isTrue(div?.classList.contains('pricing-card'));
+```
+
+Your page should contain a `pricing-card` selector.
````
```suggestion
You should have a `.pricing-card` selector.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+assert.strictEqual(h2?.textContent, 'Basic');
+```
+
+Your `basic-plan` element should contain a `p` element and the text `$9/month`.
````
```suggestion
Your `.basic-plan` element should contain a `p` element and the text `$9/month`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+assert.strictEqual(p?.textContent, '$9/month');
+```
+
+Your `pro-plan` element should contain a `h2 element` and the text `pro`.
````
```suggestion
Your `.pro-plan` element should contain a `h2` element that has the text `Pro`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+assert.strictEqual(h2?.textContent, 'Pro');
+```
+
+Your `pro-plan` element should contain a `p` element and the text `$19/month`.
````
```suggestion
Your `.pro-plan` element should contain a `p` element with the text `$19/month`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,414 @@
+
+assert.strictEqual(div?.tagName, 'DIV');
+assert.isTrue(div?.classList.contains('basic-plan'));
+assert.isTrue(div?.classList.contains('pricing-card'));
```
```suggestion
assert.exists(document.querySelector('div.pricing-card.basic-plan'));
```

the element will contain the classes by default considering the selector

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+assert.strictEqual(h2?.textContent, 'Basic');
+```
+
+Your `basic-plan` element should have a `p` element and the text `$9/month`.
````
```suggestion
Your `.basic-plan` element should have a `p` element with the text `$9/month`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+assert.strictEqual(p?.textContent, '$19/month');
+```
+
+Your `pro-plan` element should be the second element in the layout and have the property `order` and the value `1`.
````
```suggestion
Your `.pro-plan` element should have the property `order` with a value of `1`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+assert.strictEqual(h2?.textContent, 'Premium');
+```
+
+Your `premium-plan` element should have a `p` element and the text `$29/month`.
````
```suggestion
Your `.premium-plan` element should have a `p` element with the text `$29/month`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+
+```js
+const p = document.querySelector('.premium-plan>p');
+assert.strictEqual(p?.textContent, '$29/month');
````
```suggestion
const p = document.querySelector('.premium-plan > p');
assert.strictEqual(p?.textContent.trim(), '$29/month');
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,414 @@
+
+1. Your page should have a `h1` element with the text `Pricing Plans`.
+2. Your page should have a `div` element with the class `pricing-container`.
+    - It should have a `display` property and the `flex` value.
```
```suggestion
    - The `div` element should have a `display` property with the value of `flex`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,414 @@
+1. Your page should have a `h1` element with the text `Pricing Plans`.
+2. Your page should have a `div` element with the class `pricing-container`.
+    - It should have a `display` property and the `flex` value.
+    - It should have a `flex-wrap` property and the `wrap` value.
```
```suggestion
    - The `div` element should have a `flex-wrap` property with the value of `wrap`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+
+```js
+const div = document.querySelector('.pricing-container');
+assert.strictEqual(div?.tagName, 'DIV');
````
```suggestion
assert.exists(document.querySelector('h1 + div.pricing-container'));
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+assert.isTrue(div?.classList.contains('pricing-card'));
+```
+
+Your page should have a `pricing-card` selector.
````
```suggestion
You should have a `.pricing-card` selector.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+assert.strictEqual(new __helpers.CSSHelp(document).getStyle('.pricing-card')?.justifyContent, 'space-between');
+```
+
+Your `.pricing-card` selector should have the property `flex` and the value `0 0 200px`.
````
```suggestion
Your `.pricing-card` selector should have the property `flex` with a value of `0 0 200px`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,414 @@
+assert.strictEqual(h2?.textContent, 'Pro');
+```
+
+Your `pro-plan` element should have a `p` element and the text `$19/month`.
````
```suggestion
Your `.pro-plan` element should have a `p` element with the text `$19/month`.
```

majestic-owl448 reviewed (commented):
The comments I left consider using `CSSHelp`, a different approach that would be less strict about which selector to use would be using `getComputedStyle` https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle

I don't know if maybe that would be a preferred approach. It would keave more freedom of implementation.

majestic-owl448 reviewed (commented):
The comments I left consider using `CSSHelp`, a different approach that would be less strict about which selector to use would be using `getComputedStyle` https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle

I don't know if maybe that would be a preferred approach. It would keave more freedom of implementation.

ilenia-magoni commented:
GitHub is rebelling, I did not press the button to send the review...

mtrivera committed:
refactor(curriculum): Apply suggestions from code review

Clean up grammar hints and improve readability

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

Dario-DC commented on client/src/pages/learn/full-stack-developer/lab-pricing-plans-layout/index.md:
```diff
@@ -0,0 +1,9 @@
+
+## Introduction to the Design a Pricing Plans Layout Page
+
+This page is for the Design a Pricing Plans Layout Page
```
```suggestion
In this lab, you'll use flexbox to create a common three card tier layout. You'll practice aligning elements using flexbox properties like `flex`, `flex-grow`, `order`, and more.
```

Dario-DC commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,419 @@
+**User Stories**
+
+1. Your page should have a `h1` element with the text `Pricing Plans`.
+2. Your page should have a `div` element with the class `pricing-container`.
```
```suggestion
2. Your page should have a `div` element with the class `pricing-container` below the `h1` element.
```

Dario-DC commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,419 @@
+1. Your page should have a `h1` element with the text `Pricing Plans`.
+2. Your page should have a `div` element with the class `pricing-container`.
+    - The `div` element should have a `display` property with the value of `flex`.
+    - The `div` element should have a `flex-wrap` property with the value of `wrap`.
```
I think these two lines should be replaced by a single user story:
> The `.pricing-container` element should have a `display` property with the value of `flex` and a `flex-wrap` property with the value of `wrap`.

ilenia-magoni replied:
those are so granular that are more for hints than user stories

Dario-DC commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,419 @@
+3. Your page should have three `div` elements with the class `pricing-card` to represent the pricing plans.
+    - One of your `div` elements should have the class `basic-plan` in addition to the `pricing-card` class.
+    - One of your `div` elements should have the class `pro-plan` in addition to the `pricing-card` class.
+    - One of your `div` elements should have the class `premium-plan` in addition to the `pricing-card` class.
```
I would use the class to refer to these:
> One of the `.pricing-card` elements should have the class ...

Dario-DC commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,419 @@
+7. Your `pro-plan` element should have a `p` element with the text `$19/month`.
+8. Your `premium-plan` element should have an `h2` element with the text `Premium Plan`.
+9. Your `premium-plan` element should have a `p` element with the text `$29/month`.
+10. Each of your `.pricing-card` elements should use Flexbox to stack its content in a column and justify the space between the children using `space-between`. Each element should also set the `flex` property to `0 0 200px` to give it a consistent width and prevent it from growing or shrinking in the layout. Each element should set the `border` property to `2px solid black` to see how the differe ...
```
```suggestion
10. Each of your `.pricing-card` elements should:
   - Use Flexbox to stack its content in a column and justify the space between the children using `space-between`.
   - Set the `flex` property to `0 0 200px` to give it a consistent width and prevent it from growing or shrinking in the layout.
   - Set the `border` property to `2px solid black` to see how the different cards take up space.
```

Dario-DC commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,419 @@
+8. Your `premium-plan` element should have an `h2` element with the text `Premium Plan`.
+9. Your `premium-plan` element should have a `p` element with the text `$29/month`.
+10. Each of your `.pricing-card` elements should use Flexbox to stack its content in a column and justify the space between the children using `space-between`. Each element should also set the `flex` property to `0 0 200px` to give it a consistent width and prevent it from growing or shrinking in the layout. Each element should set the `border` property to `2px solid black` to see how the differe ...
+11. The `basic-plan` element should appear first in the layout. You should use the `order` property for this.
```
```suggestion
11. The `.basic-plan` element should appear first in the layout. You should use the `order` property for this.
```

Dario-DC commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,419 @@
+9. Your `premium-plan` element should have a `p` element with the text `$29/month`.
+10. Each of your `.pricing-card` elements should use Flexbox to stack its content in a column and justify the space between the children using `space-between`. Each element should also set the `flex` property to `0 0 200px` to give it a consistent width and prevent it from growing or shrinking in the layout. Each element should set the `border` property to `2px solid black` to see how the differe ...
+11. The `basic-plan` element should appear first in the layout. You should use the `order` property for this.
+12. The `pro-plan` element should appear second in the layout. You should use the `order` property and set its `flex-grow` property to `2` so it takes up more space than the other plans.
```
```suggestion
12. The `.pro-plan` element should appear second in the layout. You should use the `order` property and set its `flex-grow` property to `2` so it takes up more space than the other plans.
```

Dario-DC commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,419 @@
+assert.lengthOf(divEls, 3);
+```
+
+One of your `div` elements should have the class `pricing-card` and the class `basic-plan`.
````
```suggestion
One of your `.pricing-card` elements should have the class `basic-plan`.
```

Dario-DC commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,419 @@
+assert.exists(document.querySelector('div.pricing-card.basic-plan'));
+```
+
+One of your `div` elements should have the class `pricing-card` and the class `premium-plan`.
````
```suggestion
One of your `.pricing-card` elements should have the class `premium-plan`.
```

mtrivera committed:
style(curriculum): Apply suggestions from code review

Clean up typos and spelling errors. Break down user story 10. Rephrase the three div pricing card elements user story.

Co-authored-by: Dario <105294544+Dario-DC@users.noreply.github.com>

mtrivera committed:
style: Resolve markdown indent for user story 10

mtrivera committed:
docs: Combine user story 2 selector requirements

mtrivera committed:
refactor: Edit pricing-container selector assert

mtrivera committed:
test: Add trim method to h1 textContent assert

mtrivera committed:
style: Add whitespace to basic plan paragraph selector

Dario-DC reviewed (commented):
You need to update your branch with `git pull upstream main`. Then, considering the recent changes, I think the `intro.json` and the `responsive-web-design-v9.json` will require an update as well.

mtrivera commented:
@ilenia-magoni 
> The comments I left consider using `CSSHelp`, a different approach that would be less strict about which selector to use would be using `getComputedStyle` https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle
> 
> I don't know if maybe that would be a preferred approach. It would keave more freedom of implementation.

Running into a weird issue. In the user's `styles.css`  I have this code:

```css
.pricing-container {
	display: flex;
}
```

However, it fails the in-browser tests but passes the `pnpm test`. 
```js
const div = document.querySelector('.pricing-container');
assert.equal(getComputedStyle(div)?.display, 'flex');
```

I tested an inline style and it passes the in-browser test:
```html
<div style="display:flex;" class="pricing-container"></div>
```

mtrivera committed:
Merge branch 'main' of https://github.com/freeCodeCamp/freeCodeCamp into pricing-plan-lab-776

mtrivera commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
```diff
@@ -0,0 +1,422 @@
+
+**User Stories**
+
+1. Your page should have a `h1` element with the text `Pricing Plans`.
```
Should it not be above `review-css-flexbox` in the `intro.json`?

```js
      "lab-page-of-playing-cards": {
        "title": "Build a Page of Playing Cards",
        "intro": [
          "In this lab, you'll use flexbox to create a webpage of playing cards.",
          "You'll practice aligning elements using flexbox properties like <code>flex-direction</code>, <code>justify-content</code>, <code>align-self</code>, and more."
        ]
      },
      "lab-pricing-plans-layout": {
        "title": "Design a Pricing Plans Layout Page",
        "intro": [
          "In this lab, you'll use flexbox to create a common three-card tier layout.",
          "You'll practice aligning elements using flexbox properties like <code>flex</code>, <code>flex-grow</code>, <code>order</code>, and more."
        ]
      },
      "review-css-flexbox": {
        "title": "CSS Flexbox Review",
        "intro": [
          "Before you're quizzed on CSS flexbox, you should review what you've learned.",
          "Open up this page to review concepts like the <code>flex-direction</code>, <code>justify-content</code>, <code>align-items</code>, <code>flex-wrap</code> properties, and more."
        ]
      },
```      

How about in the `modules` section?
```js
    },
    "modules": {
      "basic-html": "Basic HTML",
      "semantic-html": "Semantic HTML",
      "html-forms-and-tables": "Forms and Tables",
      "html-and-accessibility": "Accessibility",
      "review-html": "HTML Review",
      "computer-basics": "Computer Basics",
      "basic-css": "Basic CSS",
      "design-for-developers": "Design",
      "absolute-and-relative-units": "Absolute and Relative Units",
      "pseudo-classes-and-elements": "Pseudo Classes and Elements",
      "css-colors": "Colors",
      "styling-forms": "Styling Forms",
      "css-box-model": "The Box Model",
      "css-flexbox": "Flexbox",
      "css-typography": "Typography",
      "css-and-accessibility": "Accessibility",
      "attribute-selectors": "Attribute Selectors",
      "css-positioning": "Positioning",
      "responsive-design": "Responsive Design",
      "css-variables": "Variables",
      "css-grid": "Grid",
      "css-animations": "Animations",
      "review-css": "CSS Review",
      "lab-survey-form": "Build a Survey Form",
      "lab-page-of-playing-cards": "Build a Page of Playing Cards",
      "lab-book-inventory-app": "Build a Book Inventory App",
      "lab-technical-documentation-page": "Build a Technical Documentation Page",
      "lab-product-landing-page": "Build a Product Landing Page",
      "responsive-web-design-certification-exam": "Responsive Web Design Certification Exam"
    },
```

mtrivera committed:
chore(api): Add pricing-plans to module and update block location

mtrivera committed:
chore(api): Add pricing-plan to css-flexbox block

mtrivera committed:
chore(api): Change 'blockType to 'blockLabel'

mtrivera committed:
test: Improve wording of user story 1

mtrivera commented on client/i18n/locales/english/intro.json:
```diff
@@ -7714,6 +7721,7 @@
       "review-css": "CSS Review",
       "lab-survey-form": "Build a Survey Form",
       "lab-page-of-playing-cards": "Build a Page of Playing Cards",
+      "lab-pricing-plans-layout": "Build a Pricing Plans Layout",
```
I thought it was needed for the change @Dario-DC mentioned to me. I'll remove it on a future commit.

majestic-owl448 replied:
it is not a standalone module, so it's not needed

Dario-DC replied:
I hate this file...

mtrivera committed:
chore(api): Remove lab name from modules section

mtrivera committed:
docs: Change to '.pricing-card' element for user story 3

mtrivera committed:
test: Update most tests to use getComputedStyle

majestic-owl448 commented on curriculum/structure/superblocks/full-stack-developer.json:
please remove the formatting changes from this file

Dario-DC commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,422 @@
+You should have a `.pricing-container` selector.
+
+```js
+assert.exists(document.querySelector('.pricing-container'));
````
```suggestion
assert.exists(new __helpers.CSSHelp(document).getStyle('.pricing-container'));
```

mtrivera committed:
Merge branch 'main' of https://github.com/freeCodeCamp/freeCodeCamp into pricing-plan-lab-776

mtrivera committed:
style(api): Revert formatting changes

mtrivera committed:
Apply suggestions from code review

Co-authored-by: Dario <105294544+Dario-DC@users.noreply.github.com>

mtrivera committed:
Merge branch 'pricing-plan-lab-776' of github.com:mtrivera/freeCodeCamp into pricing-plan-lab-776

majestic-owl448 commented on client/src/pages/learn/full-stack-developer/lab-pricing-plans-layout/index.md:
```diff
@@ -0,0 +1,9 @@
+---
+title: Introduction to the Design a Pricing Plans Layout Page
+block: lab-pricing-plans-layout
+superBlock: full-stack-developer
```
```suggestion
superBlock: responsive-web-design-v9
```

majestic-owl448 commented on curriculum/structure/superblocks/full-stack-developer.json:
```diff
@@ -179,6 +179,7 @@
             "workshop-flexbox-photo-gallery",
             "lab-page-of-playing-cards",
+            "lab-pricing-plans-layout",
             "review-css-flexbox",
```
```suggestion
            "lab-page-of-playing-cards",
            "review-css-flexbox",
```

we are not adding to fsd anymore

majestic-owl448 committed:
Apply suggestions from code review

majestic-owl448 committed:
index.md in right place

Dario-DC committed:
Update curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md

Dario-DC committed:
replace getComputedStyle

Dario-DC commented:
I've just pushed a commit to replace `getComputedStyle`. It should work as expected now.

majestic-owl448 commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,421 @@
+assert.strictEqual(new __helpers.CSSHelp(document).getStyle('.basic-plan')?.order, '0');
+```
+
+Your `.basic-plan` element should have a `h2` element with the text `Basic`.
````
```suggestion
Your `.basic-plan` element should have an `h2` element with the text `Basic`.
```

majestic-owl448 commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,421 @@
+assert.strictEqual(p?.textContent.trim(), '$9/month');
+```
+
+Your `.pro-plan` element should have a `h2` element with the text `Pro`.
````
```suggestion
Your `.pro-plan` element should have an `h2` element with the text `Pro`.
```

majestic-owl448 commented on curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md:
````diff
@@ -0,0 +1,421 @@
+assert.strictEqual(new __helpers.CSSHelp(document).getStyle('.pro-plan')?.flexGrow, '2');
+```
+
+Your `.premium-plan` element should have a `h2` element with the text `Premium`.
````
```suggestion
Your `.premium-plan` element should have an `h2` element with the text `Premium`.
```

majestic-owl448 committed:
Apply suggestions from code review

majestic-owl448 reviewed (approved):
Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 🎉
